Dockerfile

```diff
@@ -11,84 +11,53 @@ ENV DEBIAN_FRONTEND=noninteractive
 
 RUN apt-get -y update
 RUN apt-get -y install \
-    sudo \
     apt-utils \
-    git \
-    curl \
-    vim \
-    unzip \
-    wget \
+    automake \
     build-essential \
     cmake \
+    curl \
+    git \
+    libcupti-dev \
     libopenblas-dev \
-    automake \
+    lsof \
     openssh-client \
     openssh-server \
-    lsof \
-    python3.6 \
+    python3 \
     python3-dev \
     python3-pip \
     python3-tk \
-    libcupti-dev
+    sudo \
+    unzip \
+    vim \
+    wget
 RUN apt-get clean
 RUN rm -rf /var/lib/apt/lists/*
 
-#
-# generate python script
-#
 RUN ln -s python3 /usr/bin/python
 
 #
-# update pip
+# Install libraries
 #
-RUN python3 -m pip install --upgrade pip==20.2.4 setuptools==50.3.2
-
-# numpy 1.19.5  scipy 1.5.4
-RUN python3 -m pip --no-cache-dir install numpy==1.19.5 scipy==1.5.4
-
-#
-# TensorFlow
-#
-RUN python3 -m pip --no-cache-dir install tensorflow==2.3.1
-
-#
-# Keras
-#
-RUN python3 -m pip --no-cache-dir install Keras==2.4.3
-
-#
-# PyTorch
-#
-RUN python3 -m pip --no-cache-dir install torch==1.7.1 torchvision==0.8.2 pytorch-lightning==1.3.3
-
-#
-# sklearn 0.24.1
-#
-RUN python3 -m pip --no-cache-dir install scikit-learn==0.24.1
-
-#
-# pandas==0.23.4 lightgbm==2.2.2
-#
-RUN python3 -m pip --no-cache-dir install pandas==1.1 lightgbm==2.2.2
+RUN python3 -m pip install --upgrade pip==21.1.3 setuptools==57.4.0 wheel==0.36.2
+RUN python3 -m pip --no-cache-dir install numpy==1.21.1 scipy==1.7.0
+RUN python3 -m pip --no-cache-dir install tensorflow==2.4.2
+RUN python3 -m pip --no-cache-dir install torch==1.7.1 torchvision==0.8.2 pytorch-lightning==1.3.8
+RUN python3 -m pip --no-cache-dir install scikit-learn==0.24.2
+RUN python3 -m pip --no-cache-dir install pandas==1.3.0 lightgbm==3.2.1
+RUN python3 -m pip --no-cache-dir install azureml azureml-sdk
 
 #
 # Install NNI
 #
 COPY dist/nni-${NNI_RELEASE}-py3-none-manylinux1_x86_64.whl .
 RUN python3 -m pip install nni-${NNI_RELEASE}-py3-none-manylinux1_x86_64.whl
 
-# 
+#
```

Comment:
I think you no longer need this if you have upgraded pytorch and torchvision to the latest version.
But never mind. We can do this in future...

Reply:
If we upgrade torch to latest version, then the CUDA version is too high for tensorflow.